exporter/opentelemetry-exporter-otlp/src/opentelemetry/exporter/otlp/exporter.py

```diff
@@ -54,9 +54,31 @@
 ExportServiceRequestT = TypeVar("ExportServiceRequestT")
 ExportResultT = TypeVar("ExportResultT")
 
+_ENVIRON_TO_COMPRESSION = {
+    None: None,
+    "gzip": Compression.Gzip,
+    "deflate": Compression.Deflate,
```

Comment:
Is deflate defined in the specs?

Reply:
[Relevant part of spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#user-content-configuration-options:~:text=Supported%20values%20for%20OTEL_EXPORTER_OTLP_*COMPRESSION%20options%3A,be%20documented%20for%20each%20particular%20language.):

> Supported values for `OTEL_EXPORTER_OTLP_*COMPRESSION` options:
> 
> * If the value is missing, then compression is disabled.
> * `gzip` is the only specified compression method for now. Other options MAY be supported by language SDKs and should be documented for each particular language.

That said, I checked and the gRPC Go does not seem to support `Deflate`. I think gRPC will negotiate compression between client and server, but happy to remove for now

Reply:
Wdyt?

Reply:
If Otel Collector supports it, we should as well. If it doesn't, we can still support if it doesn't add too much maintenance overhead.

Reply:
I dont think it explicitly supports it. Based on what I read, I think the collector would re-negotiate a different compression, although I haven't tested.

I'm going to remove or now as its quite easy to add but difficult to remove later on :)

Reply:
Removed